lua/lspsaga/codeaction/indicator.lua

```diff
@@ -68,21 +68,20 @@ M.check = function()
   local fb = vim.bo.filetype
 
   if M.servers[current_file] == nil then
-    local clients = vim.lsp.get_active_clients()
-
-    for _, client in ipairs(clients) do
-      if
-        is_file
-        and client
-        and client.config.filetypes
-        and vim.tbl_contains(client.config.filetypes, fb)
-        and client.resolved_capabilities.code_action
-        and client.supports_method "code_action"
-      then
-        M.servers[current_file] = true
-        break
+    vim.lsp.for_each_buffer_client(vim.api.nvim_get_current_buf(), function(client)
+        if M.servers[current_file] then return end
+        if
+          is_file
```

Comment:
I believe this must be outside the loop. I mean maybe better at line 69. if not is_file then set M.servers[current_file] = false and return .Because If we don't have file then it does make sense to doing the rest

Reply:
Also perhaps getting active lsp doesn't make sense if we want to exit if no file, maybe move active,_ = lib.check_lsp_active to line 91

Reply:
> if not is_file then set M.servers[current_file] = false and return

Done

Reply:
> Also perhaps getting active lsp doesn't make sense if we want to exit if no file, maybe move active,_ = lib.check_lsp_active to line 91

I've done some refactoring about it. I think now it's even better. And now it is able to attach to lsp even if it is not active at file opening